### PR TITLE
chore: disallow multiple localhost connections

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

Adds basic checks to clientID on `MsgConnectionOpenInit` and `MsgConnectionOpenTry` basic validation.

closes: #3167 

### Commit Message / Changelog Entry

**NA**

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by damiannolan** [comment]
NOTE: this is only added to INIT and TRY as the client ID isn't available directly on ACK and CONFIRM msgs. It is probably not required on the TRY msg checks but I think its no harm to add it.

File: modules/core/03-connection/types/msgs.go [hunk 1]
```diff
@@ -102,6 +106,10 @@ func NewMsgConnectionOpenTry(
 
 // ValidateBasic implements sdk.Msg
 func (msg MsgConnectionOpenTry) ValidateBasic() error {
+	if msg.ClientId == exported.Localhost {
+		return sdkerrors.Wrap(clienttypes.ErrInvalidClientType, "localhost connetion handshakes are disallowed")
+	}
```
Comment (colin-axner) [inline modules/core/03-connection/types/msgs.go:111]: I think this isn't strictly necessary as there should not exist a connection in INIT which would have a valid consensus state stored for the localhost. But, I think it makes sense to be clear/defensive

**Review by colin-axner — APPROVED** [review]
Quick one! Great job!

File: modules/core/03-connection/types/msgs.go (hunk 1 above)
Comment (damiannolan) [reply modules/core/03-connection/types/msgs.go:111]: Yeah, exactly! This was my thinking too :)

**Review by chatton — APPROVED** [review]
LGTM! Nice job 👍

**Pull request merged**